### PR TITLE
Allow to release as RC

### LINKED ISSUE #42: Allow to release as RC
Idea: 
- Similar to how renovate uses checkboxes in issue / PR descriptions to re-run / start task we could use a checkbox to trigger an rc release
- alternative comment with sth like `/release rc` to trigger an rc release

Both would be done on a push run on the release branch for now later it could be possible to trigger those directly by https://github.com/woodpecker-ci/woodpecker/issues/814

### PR DESCRIPTION
**Pull request by anbraten**
closes #42 

test PR: https://github.com/anbraten/test-ready-release-go/pull/61
test release: https://github.com/anbraten/test-ready-release-go/releases/tag/1.1.0-rc.0

image

### CODE REVIEW

File: src/utils/pr.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { describe, it, expect } from "vitest";
+import { getCheckboxValueFromString } from "./pr";
+
+describe("pr", () => {
+  it("should check for the correct checkbox value", () => {
+    const prDescription = `
+This PR was opened by the [ready-release-go](https://github.com/woodpecker-ci/plugin-ready-release-go) plugin.
```
Comment (pat-s) [inline src/utils/pr.test.ts:7]: I always wonder if "ready-release-go" is the best possible name for the plugin 😅

File: src/utils/pr.test.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+When you're ready to do a release, you can merge this and a release and tag with version
```
Comment (pat-s) [inline src/utils/pr.test.ts:8]:
```suggestion
When you're ready for the next release, you can merge this PR and a release with the respective version tag
```

File: src/utils/pr.test.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+will be created automatically.If you're not ready to do a release yet, that's fine,
+whenever you add more changes to this PR will be updated.
```
Comment (pat-s) [inline src/utils/pr.test.ts:10]:
```suggestion
will be created automatically. This PR will be updated automatically for every new commit to the default branch.
```

File: src/cmd/prepare.ts [hunk 4]
```diff
@@ -106,12 +109,14 @@ export async function prepare({
...
+      `[ready-release-go](https://github.com/woodpecker-ci/plugin-ready-release-go) plugin. ` +
       `When you're ready to do a release, you can merge this and a release and tag with ` +
-      `version \`${nextVersion}\` will be created automatically.` +
+      `version \`${nextVersion}\` will be created automatically. ` +
       `If you're not ready to do a release yet, that's fine, ` +
-      `whenever you add more changes to \`${releaseBranch}\`` +
+      `whenever you add more changes to \`${releaseBranch}\` ` +
       `this PR will be updated.\n\n` +
+      `## Options\n\n` +
+      `- [${shouldBeRC ? "x" : " "}] Release this version as RC\n\n` +
```
Comment (pat-s) [inline src/cmd/prepare.ts:119]:
```suggestion
      `- [${shouldBeRC ? "x" : " "}] Mark this version as a "Release Candidate" (RC)\n\n` +
```

**Review by pat-s — COMMENTED** [review]
Some wording suggestions :)

File: src/utils/pr.test.ts (hunk 1 above)
Comment (anbraten) [reply src/utils/pr.test.ts:7]: Dunno, isn't it fitting. You wait til it's ready, a release will be created and you are ready to go with using the new version 😉

File: src/utils/pr.test.ts (hunk 2 above)
Comment (anbraten) [reply src/utils/pr.test.ts:8]: That's just a text for the test. There are some gaps as I removed the actual values while copying it from this repo.

File: src/utils/pr.test.ts (hunk 3 above)
Comment (anbraten) [reply src/utils/pr.test.ts]: just the test missing some values.

File: src/utils/pr.test.ts (hunk 2 above)
Comment (anbraten) [reply src/utils/pr.test.ts]: I've added some sample values for the gaps.

File: src/cmd/prepare.ts (hunk 4 above)
Comment (anbraten) [reply src/cmd/prepare.ts]: adjusted the text (just without `(RC)`) to keep the regex simple.

File: src/utils/pr.test.ts (hunk 1 above)
Comment (pat-s) [reply src/utils/pr.test.ts:7]: Well kind of, if you explain it like this sure. So far I just didn't read it like "ready -> release -> go" but couldn't really wrap my head around it 😅 I guess I would have simply named it `plugin-release-helper`. But then you have a broader scope than just WP so...

File: src/utils/pr.test.ts [hunk 5]
```diff
@@ -5,13 +5,13 @@ describe("pr", () => {
   it("should check for the correct checkbox value", () => {
     const prDescription = `
 This PR was opened by the [ready-release-go](https://github.com/woodpecker-ci/plugin-ready-release-go) plugin.
-When you're ready to do a release, you can merge this and a release and tag with version
-will be created automatically.If you're not ready to do a release yet, that's fine,
-whenever you add more changes to this PR will be updated.
+When you're ready to do a release, you can merge this and a new release with version '1.2.3'
+will be created automatically. If you're not ready to do a release yet, that's fine,
```
Comment (pat-s) [inline src/utils/pr.test.ts:9]: The "that's fine" still sounds a bit strange to me as it sounds a bit like talking to a child 😅 But no big deal overall...

**Review by pat-s — APPROVED** [review]

File: src/utils/pr.test.ts (hunk 5 above)
Comment (anbraten) [reply src/utils/pr.test.ts:9]: Funny, yeah now as you say that. I just copied it from the changeset bot, but guess we leave it for now.

**Pull request merged**